server/cluster/cluster.go

```diff
@@ -1781,3 +1784,42 @@ func newCacheCluster(c *RaftCluster) *cacheCluster {
 		stores:      c.GetStores(),
 	}
 }
+
+// Evict the leaders when the store is damaged. Damaged regions are emergency errors
+// and requires user to manually remove the `evict-leader-scheduler` with pd-ctl
+func (c *RaftCluster) handleDamagedStore(stats *pdpb.StoreStats) {
+	// ToDO: regions have no special process for the time being
+	// and need to be removed in the future
+	damaged_regions := stats.GetDamagedRegionsId()
+	if len(damaged_regions) == 0 {
+		return
+	}
+
+	log.Warn("store has damaged regions",
+		zap.Uint64("store-id", stats.GetStoreId()),
+		zap.Uint64s("region-ids", damaged_regions))
+	if exist, err := c.IsSchedulerExisted(schedulers.EvictLeaderName); !exist {
```

Comment:
Another question is that if the `evict-leader-scheduler` exists, we may still need to update it because it can work on different stores. This is done by updating the scheduler config. You can check the implementation of the API to see a similar logic.

https://github.com/tikv/pd/blob/9bbd6df243b483ac998fb41448d6d71c7dc42eb8/server/api/scheduler.go#L257-L263

Reply:
Finished by move `RedirectSchedulerUpdate` from`api/handler` to `handler`, which allows us to use `server.handler.RedirectSchedulerUpdate`